### PR TITLE
* Adds the 'cache-output-file` configuration option.

### PR DESCRIPTION
**Pull request by pcj**
* If defined, the `plovr serve` command will write the
  output of a successful compilation run to this file.
  Subsequent GET requests to the serve command will respond
  with the contents of this file rather than attempt
  recompile.  A recompile will be triggered if any compile
  dependencies are modified.

### CODE REVIEW

File: src/org/plovr/ConfigOption.java [hunk 1]
```diff
@@ -272,6 +272,16 @@ public void apply(String outputCharset, Config.Builder builder) {
     }
   }),
 
+
+  CACHE_OUTPUT_FILE("cache-output-file", new ConfigUpdater() {
```
Comment (nicks) [inline src/org/plovr/ConfigOption.java]: is there any reason not to make this behavior the default? Presumably plovr could just pick a tmpdir itself.

File: src/org/plovr/FileUtil.java [hunk 2]
```diff
@@ -27,4 +27,35 @@ public static boolean contains(File parent, File child) {
 
     return false;
   }
+
+
+  /**
+   * Return true if the file argument is non-null and has a
+   * last-modified date greater than the given lastModified time.
+   *
+   * @author pcj@pubref.org (Paul Johnston)
```
Comment (nicks) [inline src/org/plovr/FileUtil.java]: it seems a bit weird to author-annotate individual functions

Comment (pcj) [inline src/org/plovr/FileUtil.java]: Such an awesome function though!  (can remove it).

File: src/org/plovr/ConfigOption.java (hunk 1 above)
Comment (nicks) [inline src/org/plovr/ConfigOption.java]: maybe the answer is that it uses a tmp file by default, but allows you to set a cache-output-file if you care

or maybe cache-output-file is useful for cases where you plan to start and stop plovr but want the cache to persist.

Comment (pcj) [inline src/org/plovr/ConfigOption.java]: I'm fine making it the default behavior in which case there could be a opt-out option to disable it.  I made it opt-in in the case there were unintended side-effects, but I don't see any at this time.

**Comment by nicks** [comment]
thanks!

**Comment by nicks** [comment]
fyi -- i noticed there's a minor bug here. Plovr allows you to set certain compilation parameters from the request query parameters (e.g., ?mode=ADVANCED), and if you do that with a cache-output-file, it will use the cached version with the wrong mode.

I refactored the code a bit to use an in-memory cache by default (which invalidates correctly). I left a comment about the disk invalidation problem but i'm not sure how to handle that yet...

**Pull request merged**